### DISCUSSION TITLE
When is data from a query with Suspense undefined?

### DISCUSSION THREAD

**Discussion by iway1** [TanStack/query#3606] [Q&A]
Getting confused about the behavior of queries with Suspense turned on. I assumed that "data" is guaranteed to be defined with suspense, but I'm running into issues where `data` is undefined after invalidating queries, even though my logs tell me that the data is refetched before the error happens?

Anyone have any idea of what are the circumstances under which `data` will be undefined when `suspense: true`? Or maybe this is a bug? Doesn't make sense that my data would be refetched and then it would be undefined in the component and crash after the fetch.

**Comment by TkDodo** [comment]
Can you show this behaviour in a codesandbox reproduction? I would assume that data is also not undefined. That is, unless you have `enabled: false` for example.

Since you mention "after invalidating queries": my guess would be that for some reason, the API might not return what you expect, which will somehow make you return `undefined` from the `queryFn`. Is the query still in `success` state?
